addons/project/models/project_task.py

```diff
@@ -779,6 +780,13 @@ def _inverse_display_name(self):
                         extract_data(task)
                 task.name = task.display_name.strip()
 
+    def _compute_link_preview_name(self):
+        for task in self:
+            if task.project_id.id:
+                task.link_preview_name = task.display_name + ' | ' + task.project_id.name
+            else:
+                task.link_preview_name = task.display_name
```

Comment:
Seems equivalent to:
```suggestion
    def _compute_link_preview_name(self):
        for task in self:
            link_preview_name = task.display_name
            if task.project_id:
                link_preview_name += f' | {task.project_id.name}'
            task.link_preview_name = link_preview_name
```
Also, the user can have access to the task but no always to the project. So, we should add a sudo to get the project name.

```suggestion
    def _compute_link_preview_name(self):
        for task in self:
            link_preview_name = task.display_name
            if task.project_id:
                link_preview_name += f' | {task.project_id.sudo().name}'
            task.link_preview_name = link_preview_name
```